crates/runtime/src/memfd.rs

```diff
@@ -428,6 +437,7 @@ impl MemFdSlot {
         // The initial memory image, if given. If not, we just get a
         // memory filled with zeroes.
         if let Some(image) = maybe_image {
+            assert!(image.offset.checked_add(image.len).unwrap() <= initial_size_bytes);
             if image.len > 0 {
                 let image = image.clone();
```

Comment:
Something I just realized for this case, I think that `self.image` should be set to `None` if we don't take this path through the `if`, right?

Reply:
Eek, indeed. This "oh wait one more case" on the core code is making me nervous enough that I'm gonna go figure out how to test this more thoroughly (so far it's been just "every other test uses this as a backend").

Reply:
Internally it might be a bit more robust to use `Range<usize>` instead of `start: usize, len: usize` perhaps, but otherwise I don't think this was a functionality bug since it's more that it could theoretically trip the preexisting assert in setting page protections.

Reply:
oh oops I thought I was responding to https://github.com/bytecodealliance/wasmtime/pull/3697#discussion_r797879911 here, but I clearly am not, so disregard that last comment here!

Reply:
Ah yes, I see now; pushed an update that modifies `set_protection` to use `Range` instead and this is indeed much clearer!